go/ct/common/u256_test.go

```diff
@@ -461,3 +462,178 @@ func TestU256_MarshallingRoundTrip(t *testing.T) {
+		if want, got := test.original, test.converted.Uint256(); !want.Eq(&got) {
+			t.Errorf("Unexpected result from Uint256, want %v, got %v", want, got)
+		}
+		if want, got := test.converted, NewU256FromUint256(test.original); !want.Eq(got) {
+			t.Errorf("Unexpected result from NewU256FromUint256, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_SignExtend(t *testing.T) {
+	tests := []struct {
+		value U256
+		want  U256
+	}{
+		{U256{}, U256{}},
+		{NewU256(1), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff)},
+		{NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff)},
+		{NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff)},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.value.SignExtend(NewU256(0xfff)); !want.Eq(got) {
+			t.Errorf("Unexpected result from SignExtend, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_And(t *testing.T) {
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(1), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff), NewU256(0xff)},
+		{NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff)},
+		{NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff)},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.a.And(test.b); !want.Eq(got) {
+			t.Errorf("Unexpected result from And, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Or(t *testing.T) {
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(1), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff), NewU256(0xff)},
+		{NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff)},
+		{NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff)},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.a.Or(test.b); !want.Eq(got) {
+			t.Errorf("Unexpected result from Or, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Xor(t *testing.T) {
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(1), U256{}},
+		{NewU256(0xff), NewU256(0xff), U256{}},
+		{NewU256(0xffffffffffffffff), NewU256(0xffffffffffffffff), U256{}},
+		{NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), U256{}},
```

Comment:
This test would be passed by an `Xor` that just always resets the content of the value to zero.